core/src/main/java/io/micronaut/core/annotation/AnnotationDefaultValuesProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.annotation;
+
+import java.util.Map;
+
+/**
+ * The annotation default values provider.
+ *
+ * @author Denis Stepanov
+ * @since 4.3.0
+ */
+public interface AnnotationDefaultValuesProvider {
+
+    /**
+     * @param annotationName The annotation name
+     * @return The annotation default values
+     */
+    @NonNull
+    Map<CharSequence, Object> provide(String annotationName);
```

Comment:
Why CharSequence instead of String?

CS doesn't have a well-defined eq/hc

Reply:
Not really sure, we have it everywere

Reply:
`CharSequence` is parent of both Java's `String` and Groovy's `GString`.

Reply:
that means we need it for the processor i guess, but maybe we can move to String on the runtime side